File: Python3/1684_count_the_number_of_consistent_strings.py
# https://leetcode.cn/problems/count-the-number-of-consistent-strings/

class Solution:
    def countConsistentStrings(self, allowed, words):
        """
        Inputs:
            allowed: str
            words: list[str]
        Outputs:
            res: int
        """
        res = len(words)
        table = [0 for _ in range(26)]
        for ch in allowed:
            table[ord(ch) - ord("a")] = 1
        for word in words:
            for ch in word:
                if table[ord(ch) - ord("a")] != 1:
                    res -= 1
                    break
        return res

File: Cpp/1684_count_the_number_of_consistent_strings.cpp
// https://leetcode.cn/problems/count-the-number-of-consistent-strings/

class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int res = words.size();
        vector<int> table(26, 0);
        for (char ch: allowed) {
            table[ch - 'a'] = 1;
        }
        for (string& word: words) {
            for (char ch: word) {
                if (table[ch - 'a'] == 0) {
                    res -= 1;
                    break;
                }
            }
        }
        return res;
    }
};
